source/docs/advanced/server/configurations.md

```diff
@@ -0,0 +1,246 @@
+
+```js
+const name = "serviceProvider"
+```
+
+### Schema (key `schema`, optional)
+
+It represents the definition of the fields that will appear in the global configuration object if the configuration provider is registered. It can define things like field formats, default values or environment variables.
+
+The schema definition is based on [convict](https://github.com/mozilla/node-convict), a library developed by Mozilla to validate configurations. For a single field, the schema will have these keys:
+
+- `doc` (string): A description of the field and pointers to learn how to get its value if it requires a manual operation
+- `format` (string, array or function): The formatter used for this field's value. See [how validation works in convict](https://github.com/mozilla/node-convict#validation).
+- `default` (any): A default value. If you don't have a default value, you still have to set one by using the `null`. If there is no default value, it won't be considered as a field definition.
+- `env` (string, optional): The name of the environment variable that should be used as the value config. If none is given, the configuration won't be configurable by environment. Please keep in mind that environment variables should still match [Front-Commerce's naming convention](/docs/reference/environment-variables.html#Add-your-own-environment-variables).
+
+Thus, if we want to define a config named `serviceKey` available in `req.config.serviceKey` that would take its value from the environment variable `FRONT_COMMERCE_SERVICE_KEY`, we would use this schema definition:
+
+```js
+const schema = {
+  serviceKey: {
+    doc: "The key to get access to our remote service",
+    format: String,
+    default: null,
+    env: "FRONT_COMMERCE_SERVICE_KEY"
+  }
+};
+```
+
+A configuration provider's schema is not limited to a single field though. You can set multiple configuration keys but also nest deeper objects. For instance, if we want to group a `key` and a `secret` into a single `service` key, we would write the following schema:
+
+```js
+const schema = {
+  service: {
+    key: {
+      doc: "The key to get access to our remote service",
+      format: String,
+      default: null,
+      env: "FRONT_COMMERCE_SERVICE_KEY"
+    },
+    secret: {
+      doc: "The secret to get access to our remote service",
+      format: String,
+      default: null,
+      env: "FRONT_COMMERCE_SERVICE_SECRET"
+    }
+  }
+};
+```
+
+You could then access the values with `req.config.service.key` or `req.config.service.secret`.
+
+Please note that if another configuration provider's schema also defined a `service` key, it would merge the definitions and in your final `req.config.service` you would have both the keys from the other configuration provider's schema and from the above schema.
+
```

Comment:
Should we mention (in a blockquote for instance) the `url-without-slash` and `collection` types  available in Front-Commerce?

Reply:
I don't think it's relevant for now.